python/cudf/cudf/_lib/pylibcudf/column.pxd

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_view cimport column_view
+from cudf._lib.cpp.types cimport offset_type, size_type
+
+from .gpumemoryview cimport gpumemoryview
+from .types cimport DataType
+
+
+cdef class Column:
+    cdef:
+        # Core data
+        DataType data_type
+        size_type size
+        gpumemoryview data
+        gpumemoryview mask
+        size_type null_count
+        offset_type offset
+        # TODO: Is there a more efficient container I could use than a list
+        # here? For now defaulting to List[gpumemoryview]
+        object children
+
+        # Internals
+        unique_ptr[column_view] _underlying
```

Comment:
Does it really matter if this is a `unique_ptr[column_view]` v/s just `column_view`?

Reply:
OK, I see your comment below (https://github.com/rapidsai/cudf/pull/13562/files#r1230200771)

Reply:
We should probably document somewhere why we're doing it this way

Reply:
Added docs.